src/pivot-table/hooks/__tests__/use-selections-model.test.ts

```diff
@@ -44,39 +52,66 @@ describe("useSelectionsModel", () => {
   });
 
   test("should select cell and call begin selection", async () => {
-    const { result } = renderHook(() => useSelectionsModel(selections));
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
 
     await act(async () => {
+      Object.entries(callbacks).forEach(([, cb]) => cb());
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 2)();
     });
 
     // eslint-disable-next-line @typescript-eslint/unbound-method
     await waitFor(() => expect(selections.begin).toHaveBeenCalled());
+    await waitFor(() => expect(selections.on).toHaveBeenCalled());
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledTimes(1));
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 2)).toBeTruthy());
   });
 
   test("should select cell and not call begin selection when already active", async () => {
     selections.isActive = () => true;
-    const { result } = renderHook(() => useSelectionsModel(selections));
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
 
     await act(async () => {
+      Object.entries(callbacks).forEach(([, cb]) => cb());
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
     });
 
     // eslint-disable-next-line @typescript-eslint/unbound-method
     await waitFor(() => expect(selections.begin).toHaveBeenCalledTimes(0));
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledTimes(1));
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
   });
 
+  test("should select cell and mimic confirm action happening by calling related callback", async () => {
+    selections.isActive = () => true;
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
+
+    await act(async () => {
+      Object.entries(callbacks).forEach(([, cb]) => cb());
+      await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
+    });
+
+    act(() => {
+      // trigger callback only if eventKey is "confirmed"
+      Object.entries(callbacks).forEach(([_evtKey, cb]) => {
+        if (_evtKey === "confirmed") cb();
+      });
+    });
```

Comment:
And here I don't see a point of first calling all callbacks, and then seperately running the confirm callback. I'd say the idea of the test is good, just that running confirm once would be more accurate to what happens in the real implementation

Reply:
Correct, since we have all event types in hashmap now, we can directly call them using their key such as `callbacks.confirmed()` and avoid loop and calling other event callbacks